File: drivers/Connector-J/src/main/java/com/actiontech/getFile.java
/*
 * Copyright (C) 2016-2019 ActionTech.
 * License: https://www.mozilla.org/en-US/MPL/2.0 MPL version 2 or higher.
 */
package com.actiontech;

import java.io.File;
import java.util.ArrayList;
import java.util.List;

public class getFile {

    public static ArrayList<String> getFiles(String path) {
        ArrayList<String> filelist = new ArrayList<String>();
        File files = new File(path);
        File[] array = files.listFiles();
        for (int i = 0; i < array.length; i++) {
            if (array[i].isFile()) {
                String sqlpathname = array[i].getAbsolutePath();
                String sqlpath = array[i].getPath();
                String sqlname = array[i].getName();
                filelist.add(sqlpathname);
            } else {
                if (array[i].isDirectory()) {
                    getFiles(array[i].getPath());
                }
            }
        }
        return filelist;
    }
}

File: drivers/Connector-.Net/netdriver/getfile.cs
/*
 * Copyright (C) 2016-2022 ActionTech.
 * License: https://www.mozilla.org/en-US/MPL/2.0 MPL version 2 or higher.
 */
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.IO;

namespace netdriver
{
    class GetFile
    {
        public static List<String> GetFiles(String[] filepath)
        {
            //if file exists
            List<String> files = new List<String>();
            for (int i = 0; i < filepath.Length; i++)
            {
                var file = filepath[i];
                Console.WriteLine(filepath[i]);
                if (!File.Exists(file))
                {
                    Console.WriteLine(filepath[i] + "  doesn't exists");
                }
                else
                {
                    files.Add(filepath[i]);
                }
            }
            return files;
        }
    }
}
